### PR TITLE
Remove dynamic bag state bloat bond

### LINKED ISSUE #3580: Remove dynamic bag state bloat bond
# Background
As we started to work on state bloat bonds for the content directory, where each channel corresponds to one and exactly one dynamic bag, it became clear that it is a bit redundant to have a bloat bond for _both_ the channel and the corresponding _dynamic bag_. It also seems obvious that all uses of dynamic bags will have this pattern where there is a separate state object in another part of the runtime that effectivel serves the same purpose.

# Proposal
Just remove bloat bonds for dynamic bags, which will simplify that pallet a good amount.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201738706333715/1202127933746305) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by L3pereira**
- Closes #3580

content-pallet is also affected

Merge after  #3746

### CODE REVIEW

File: runtime-modules/storage/src/lib.rs [hunk 1]
```diff
@@ -988,12 +965,12 @@ impl VoucherUpdate {
         }
     }
 
-    fn sub_object(self, size: u64) -> Self {
-        Self {
-            objects_number: self.objects_number.saturating_sub(1),
-            objects_total_size: self.objects_total_size.saturating_sub(size),
-        }
-    }
+    // fn sub_object(self, size: u64) -> Self {
```
Comment (dobertRowneySr) [inline runtime-modules/storage/src/lib.rs]: Please remove commented code if you think it is not necessary

File: runtime-modules/storage/src/lib.rs [hunk 2]
```diff
@@ -953,13 +930,13 @@ struct VoucherUpdate {
 }
 
 impl VoucherUpdate {
-    fn add_objects_list<Balance>(self, list: &[DataObject<Balance>]) -> Self {
-        list.iter().fold(self, |acc, obj| acc.add_object(obj.size))
-    }
+    // fn add_objects_list<Balance>(self, list: &[DataObject<Balance>]) -> Self {
```
Comment (dobertRowneySr) [inline runtime-modules/storage/src/lib.rs]: Please remove commented code if you think it is not necessary

File: runtime-modules/storage/src/lib.rs [hunk 3]
```diff
@@ -953,13 +930,13 @@ struct VoucherUpdate {
...
+    //     list.iter().fold(self, |acc, obj| acc.add_object(obj.size))
+    // }
 
-    fn sub_objects_list<Balance>(self, list: &[DataObject<Balance>]) -> Self {
-        list.iter().fold(self, |acc, obj| acc.sub_object(obj.size))
-    }
+    // fn sub_objects_list<Balance>(self, list: &[DataObject<Balance>]) -> Self {
```
Comment (dobertRowneySr) [inline runtime-modules/storage/src/lib.rs]: Please remove commented code if you think it is not necessary

**Review by dobertRowneySr — APPROVED** [review]
Just some cosmetics, otherwise LGTM

**Comment by L3pereira** [comment]
You have this PR rebased on [Storage pallet refunds based on old data object prizes](https://github.com/Joystream/joystream/pull/3746#)
#3746, the cosmetics you pointed out were solved in the that PR

**Review by dobertRowneySr — APPROVED** [review]
LGTM

**Pull request merged**